chirp/drivers/uvk5.py

```diff
@@ -0,0 +1,1339 @@
+    status.msg = "Uploaded OK"
+
+    _resetradio(serport)
+
+    return True
+
+
+def _find_band(hz):
+    mhz = hz/1000000.0
+    for a in BANDS:
+        if mhz >= BANDS[a][0] and mhz <= BANDS[a][1]:
+            return a
+    return False
+
+
+@directory.register
+class TemplateRadio(chirp_common.CloneModeRadio):
+    """Quansheng UV-K5"""
+    VENDOR = "Quansheng"
+    MODEL = "UV-K5"
+    BAUD_RATE = 38400
+
+    NEEDS_COMPAT_SERIAL = False
+    FIRMWARE_VERSION = ""
+
+    def get_prompts(x=None):
+        rp = chirp_common.RadioPrompts()
+        rp.experimental = \
+            ('This is an experimental driver for the Quanscheng UV-K5. '
+             'It may harm your radio, or worse. Use at your own risk.\n\n'
+             'Before attempting to do any changes please download'
+             'the memory image from the radio with chirp or k5prog '
+             'and keep it. This can be later used to recover the '
+             'original settings. \n\n'
+             'FM radio, DTMF settings and scanlists are not yet implemented')
+        rp.pre_download = _(
+            "1. Turn radio on.\n"
+            "2. Connect cable to mic/spkr connector.\n"
+            "3. Make sure connector is firmly connected.\n"
+            "4. Click OK to download image from device.\n\n"
+            "It will may not work if you turn o the radio "
+            "with the cable already attached\n")
+        rp.pre_upload = _(
+            "1. Turn radio on.\n"
+            "2. Connect cable to mic/spkr connector.\n"
+            "3. Make sure connector is firmly connected.\n"
+            "4. Click OK to upload the image to device.\n\n"
+            "It will may not work if you turn o the radio "
+            "with the cable already attached")
+        return rp
+
+    # Return information about this radio's features, including
+    # how many memories it has, what bands it supports, etc
+    def get_features(self):
+        rf = chirp_common.RadioFeatures()
+        rf.has_bank = False
+        rf.valid_dtcs_codes = DTCS_CODES
+        rf.has_rx_dtcs = True
+        rf.has_ctone = True
+        rf.has_settings = True
+        rf.has_comment = False
+        rf.valid_name_length = 16
+        rf.valid_power_levels = UVK5_POWER_LEVELS
+
+        # hack so we can input any frequency,
+        # the 0.1 and 0.01 steps don't work unfortunately
+        rf.valid_tuning_steps = [0.01, 0.1, 1.0] + STEPS
+
+        rf.valid_tmodes = ["", "Tone", "TSQL", "DTCS", "Cross"]
+        rf.valid_cross_modes = ["Tone->Tone", "Tone->DTCS", "DTCS->Tone",
+                                "->Tone", "->DTCS", "DTCS->", "DTCS->DTCS"]
+
+        rf.valid_characters = chirp_common.CHARSET_ASCII
+        rf.valid_modes = ["FM", "NFM", "AM"]
+        rf.valid_tmodes = ["", "Tone", "TSQL", "DTCS", "Cross"]
+
+        rf.valid_skips = [""]
+
+        # This radio supports memories 1-200, 201-214 are the VFO memories
+        rf.memory_bounds = (1, 214)
```

Comment:
This is not correct. The `memory_bounds` should be the limits of the regular numbered channels. Special channels are handled by declaring `valid_special_chans`. It's still helpful to set `mem.number` on them to these for indexing purposes, but you should not just map them into the numbered list.